compiler/test/dotty/tools/dotc/reporting/CodeActionTest.scala

```diff
@@ -52,7 +52,52 @@ class CodeActionTest extends DottyTest:
       // TODO look into trying to remove the extra space that is left behind
       """|final  class Test
          |""".stripMargin
+      )
 
+  @Test def insertMissingCases =
+    checkCodeAction(
+      code =
+        """|enum Tree:
+           |  case Node(l: Tree, r: Tree)
+           |  case Leaf(v: String)
+           |
+           |object Test:
+           |  def foo(tree: Tree) = tree match {
+           |    case Tree.Node(_, _) => ???
+           |  }
+           |""".stripMargin,
+         title = "Insert missing cases (1)",
+      expected =
+        """|enum Tree:
+           |  case Node(l: Tree, r: Tree)
+           |  case Leaf(v: String)
+           |
+           |object Test:
+           |  def foo(tree: Tree) = tree match {
+           |    case Tree.Node(_, _) => ???
+           |    case Tree.Leaf(_) => ???
+           |  }
+           |""".stripMargin,
+         afterPhase = "patternMatcher"
+      )
+
+  @Test def insertMissingCasesForUnionStringType =
+    checkCodeAction(
+      code =
+        """object Test:
+           |  def foo(text: "Alice" | "Bob") = text match {
+           |    case "Alice" => ???
+           |  }
+           |""".stripMargin,
+         title = "Insert missing cases (1)",
+      expected =
+        """object Test:
+           |  def foo(text: "Alice" | "Bob") = text match {
+           |    case "Alice" => ???
+           |    case "Bob" => ???
```

Comment:
This test fails because string literals are rendered without surrounding quotes.